### ISSUE TITLE
Explore further options for pinning Apalache release

### ISSUE THREAD

**Issue by thpani** [informalsystems/quint#1248]
We are currently hardcoding the Apalache version to download on-demand when no local or running Apalache server is detected.

@shonfeder pointed out the following, improved options on #1106. We should explore these, both as part of ADR008 and the implementation.

1. We could maintain list of supported versions in a file somewhere (e.g., in the quint repo) and just request the file content, without having to hit the GH API. We could semi-automate updates to this page.
2. We could scrape the releases from https://github.com/informalsystems/quint/releases, again only requiring content of a served page, rather then going through the API.
3. We could require users to authenticate with github in order to use the apalache managed version. This would remove the rate-limiting issue, and we could also authenticate in this way in our CI.

Note that part of the problem in the previous implementation is that, iirc, every invocation of `verify` was hitting the GH API, meaning you could not invoke the subcommand more than 60 times an hour from the same IP address. If we only checked for updated versions periodically -- e.g., once per week, or when a user runs an `update` subcommand -- we would not hit the exact same problem (but other would persist).

_Originally posted by @shonfeder in https://github.com/informalsystems/quint/pull/1106#discussion_r1391655466_

**Comment by bugarela** [comment]
@thpani and @konnov, since you are using different versions of Apalache more frequently - do you think there is still a problem to solve here? In my perspective, pinning the release has been great, but I'd appreciate your input before closing this.

**Comment by thpani** [comment]
Hi @bugarela, thanks for the ping!

This is definitely still an issue – I have to use `quint verify --apalache-version` quite often.

Even with `--apalache-version` it's quite cumbersome, because it fetches the specified version, but will revert to the pinned version if I don't remember to pass `--apalache-version` all the time.

**Comment by konnov** [comment]
> [@thpani](https://github.com/thpani) and [@konnov](https://github.com/konnov), since you are using different versions of Apalache more frequently - do you think there is still a problem to solve here? In my perspective, pinning the release has been great, but I'd appreciate your input before closing this.

Yeah, this looks quite inconvenient. We basically have to open a PR on the Quint repo any time an issue is fixed. Ideally, we should have two kinds of Apalache releases: unstable and stable. This would be similar to rust switching between two kinds of versions.
